### PR TITLE
markdown: fixed ascii encoding typo

### PR DESCRIPTION
**Pull request by 0einstein0**
:heart: Thank you for your contribution!

### Description

Closes Issue [#310](https://github.com/zenodo/ops/issues/310)

- [x] Fix default to `utf-8` in case of `ascii`
- [x] Provide a fallback message instead of displaying an internal server error

Screenshot 2024-01-25 at 14 49 20

### Checklist

Ticks in all boxes and 🟢 on all GitHub actions status checks are required to merge:

- [ ] I'm aware of the [code of conduct](https://inveniordm.docs.cern.ch/contribute/code-of-conduct/).
- [ ] I've created [logical separate commits](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commits) and followed the [commit message format](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commit-message).
- [ ] I've added relevant test cases.
- [ ] I've added relevant documentation.
- [ ] I've marked [translation strings](https://inveniordm.docs.cern.ch/develop/best-practices/i18n/) (for relevant code).
- [ ] I've followed the [CSS/JS](https://inveniordm.docs.cern.ch/develop/best-practices/css-js/) and [React](https://inveniordm.docs.cern.ch/develop/best-practices/react/) guidelines (for relevant code).
- [ ] I've followed the [web accessibility](https://inveniordm.docs.cern.ch/develop/best-practices/accessibility/) guidelines (for relevant code).
- [ ] I've followed the [user interface](https://inveniordm.docs.cern.ch/develop/best-practices/ui/) guidelines (for relevant code).
- [ ] I've identified the [copyright holder(s)](https://inveniordm.docs.cern.ch/contribute/copyright-policy/) and updated copyright headers for touched files (>15 lines contributions).
- [ ] I've NOT included third-party code (copy/pasted source code or new dependencies).

**Third-party code**

If you've added [third-party code (copy/pasted or new dependencies)](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#third-party-codedependencies), please reach out to an [architect](https://github.com/orgs/inveniosoftware/teams/architects/members).

**Reminder**

By using GitHub, you have already agreed to the [GitHub’s Terms of Service](https://help.github.com/articles/github-terms-of-service/#6-contributions-under-repository-license) including that:

1. You license your contribution under the same terms as the current repository’s license.
2. You agree that you have the right to license your contribution under the current repository’s license.

### CODE REVIEW

**Review by jrcastro2 — APPROVED** [review]
Reviewed with @yashlamba

File: invenio_previewer/extensions/mistune.py [hunk 1]
```diff
@@ -22,8 +22,12 @@ def render(file):
     """Render HTML from Markdown file content."""
     with file.open() as fp:
         encoding = detect_encoding(fp, default="utf-8")
-        result = mistune.markdown(fp.read().decode(encoding))
-        return result
+        try:
+            result = mistune.markdown(fp.read().decode(encoding))
+            return result
+        except Exception as e:
```
Comment (alejandromumo) [inline invenio_previewer/extensions/mistune.py:28]: nit: I would move the error handling to the caller. Then the caller can decide what to do with it (i.e. log an error, render a default)

File: invenio_previewer/extensions/mistune.py [hunk 2]
```diff
@@ -33,10 +37,14 @@ def can_preview(file):
...
+    if content is not None:
+        return render_template(
+            "invenio_previewer/mistune.html",
+            file=file,
+            content=content,
+            js_bundles=current_previewer.js_bundles,
+            css_bundles=current_previewer.css_bundles,
+        )
+    else:
+        return render_template("invenio_previewer/default.html")
```
Comment (zzacharo) [inline invenio_previewer/extensions/mistune.py:50]:
```suggestion
    try:
        content = render(file)
        return render_template(
            "invenio_previewer/mistune.html",
            file=file,
            content=content,
            js_bundles=current_previewer.js_bundles,
            css_bundles=current_previewer.css_bundles,
        )
    except Exception as e:
        current_app.logger.exception(str(e))
        return render_template("invenio_previewer/default.html")
```

File: invenio_previewer/extensions/mistune.py [hunk 3]
```diff
@@ -22,8 +22,12 @@ def render(file):
...
     with file.open() as fp:
         encoding = detect_encoding(fp, default="utf-8")
-        result = mistune.markdown(fp.read().decode(encoding))
-        return result
+        try:
+            result = mistune.markdown(fp.read().decode(encoding))
+            return result
+        except Exception as e:
+            current_app.logger.exception(str(e))
+            return None
```
Comment (zzacharo) [inline invenio_previewer/extensions/mistune.py:30]:
```suggestion
        return mistune.markdown(fp.read().decode(encoding))
```

**Review by zzacharo — APPROVED** [review]

**Pull request merged**